java/maven.indexer/src/org/netbeans/modules/maven/indexer/NexusRepositoryIndexerImpl.java

```diff
@@ -359,24 +365,21 @@ private boolean loadIndexingContext(final RepositoryInfo info) throws IOExceptio
             }
             LOGGER.log(Level.FINE, "Loading Context: {0}", info.getId());
 
-            List<IndexCreator> creators = new ArrayList<>();
-            try {
-                for (IndexCreator creator : embedder.lookupList(IndexCreator.class)) {
-                    if (OsgiArtifactIndexCreator.ID.equals(creator.getId())) {
-                        continue; //we are no interested in osgi related content in lucene documents or ArtifactInfo objects.
-                        //they take up a lot of memory and we never query them AFAIK. (import/export packages can take up to 300k
-                        //239915, 240150 + according to my knowledge we don't expose any api that would allow 3rd party plugins to query the osgi stuff
-                    }
-                    creators.add(creator);
-                }
-            } catch (ComponentLookupException x) {
-                throw new IOException(x);
-            }
-            if (info.isLocal()) { // #164593
-                creators.add(new ArtifactDependencyIndexCreator());
-                creators.add(new ClassDependencyIndexCreator());
+            List<IndexCreator> creators;
+            if (info.isLocal()) {
+                creators = List.of(
+                    new JarFileContentsIndexCreator(),
+                    new MinimalArtifactInfoIndexCreator(),
+                    new MavenArchetypeArtifactInfoIndexCreator(),
+                    new MavenPluginArtifactInfoIndexCreator(),
+                    new ArtifactDependencyIndexCreator(),
+                    new ClassDependencyIndexCreator()
+                );
             } else {
-                creators.add(new NotifyingIndexCreator());
+                creators = List.of(
+                    new MinimalArtifactInfoRemoteIndexCreator(),
+                    new NotifyingIndexCreator()
+                );
```

Comment:
I understand, that `MavenPluginArtifactInfoIndexCreator` and `MavenArchetypeArtifactInfoIndexCreator` are dropped in addition to the `JarFileContentsIndexCreator` and `OsgiArtifactIndexCreator`, as both need the artifact and thus won't work for remote repositories?

Reply:
correct
 - OsgiArtifactIndexCreator was already filtered out before
 - JarFileContentsIndexCreator, MavenArchetypeArtifactInfoIndexCreator, MavenPluginArtifactInfoIndexCreator, ArtifactDependencyIndexCreator and ClassDependencyIndexCreator are all for the local .m2 repo, not for remote indices.